rm-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/rm/server/NSCreationServlet.java

```diff
@@ -159,55 +160,71 @@ private void createNs(HttpServletRequest request, HttpServletResponse response)
-                                                                                    toArray(policyFileParams),
-                                                                                    nodesRecoverable);
+                jsonResponsePayload = ((RMServiceImpl) RMServiceImpl.get()).defineNodeSource(sessionId,
+                                                                                             nsName,
+                                                                                             infra,
+                                                                                             toArray(infraParams),
+                                                                                             toArray(infraFileParams),
+                                                                                             policy,
+                                                                                             toArray(policyParams),
+                                                                                             toArray(policyFileParams),
+                                                                                             nodesRecoverable);
             }
 
-            if (deployNodeSource) {
-                jsonResult = ((RMServiceImpl) RMServiceImpl.get()).deployNodeSource(sessionId, nsName);
+            JSONObject jsonObject = new JSONObject(jsonResponsePayload);
+            if (hasResult(jsonObject) && isResultSuccessful(jsonObject)) {
+                jsonResponsePayload = createJsonPair(jsonObject);
+            } else {
+                deployNodeSource = false;
             }
 
-            JSONObject json = new JSONObject(jsonResult);
-            if (json != null) {
-                if (json.has("result")) {
-                    if (json.getBoolean("result")) {
-                        jsonResult = createNonEscapedSimpleJsonPair("result", "true");
-                    } else {
-                        String errorMessage = json.get("errorMessage").toString();
-                        write(response,
-                              createJavascriptPayload(callbackName,
-                                                      createEscapedSimpleJsonPair("errorMessage", errorMessage)));
-                        return;
+            if (deployNodeSource) {
+                jsonResponsePayload = ((RMServiceImpl) RMServiceImpl.get()).deployNodeSource(sessionId, nsName);
 
-                    }
+                jsonObject = new JSONObject(jsonResponsePayload);
+                if (hasResult(jsonObject)) {
+                    jsonResponsePayload = createJsonPair(jsonObject);
                 }
             }
-            write(response, createJavascriptPayload(callbackName, jsonResult));
+
+            write(response, createJavascriptPayload(callbackName, jsonResponsePayload));
         } catch (Throwable t) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Catch Exception instead of Throwable. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1181)